app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.java

```diff
@@ -122,6 +126,14 @@ public static BrowserFragment create(String url) {
     @Override
     public void onPause() {
         super.onPause();
+        getContext().unregisterReceiver(DownloadReceiver.getDownloadReceiver());
+    }
+
+    @Override
+    public void onResume(){
+        super.onResume();
+        IntentFilter intentFilter = new IntentFilter(DownloadManager.ACTION_DOWNLOAD_COMPLETE);
+        getContext().registerReceiver(DownloadReceiver.getDownloadReceiver(),intentFilter);
```

Comment:
What if any other app sends an intent to your ACTION_DOWNLOAD_COMPLETE? Can you limit it to only allow intent from this app only? How about using LocalBroadcastReceiver?